synapse/lib/stormlib/aha.py

```diff
@@ -1,22 +1,105 @@
+         'type': {'type': 'function', '_funcname': '_methSvcDel',
+                  'args': (
+                      {'name': 'name', 'type': 'str', 'desc': 'The name of the service to delete.', },
+                      {'name': 'network', 'type': 'str', 'desc': 'An optional network to use when looking up the service name.',
+                       'default': None},
+                  ),
+                  'returns': {'type': 'null'}}},
+        {'name': 'get', 'desc': 'Get information about an AHA service.',
+         'type': {'type': 'function', '_funcname': '_methSvcGet',
+                  'args': (
+                      {'name': 'name', 'type': 'str', 'desc': 'The name of the AHA service to look up.', },
+                      {'name': 'filters', 'type': 'dict', 'desc': 'An optional dictionary of filters to use when resolving the AHA service.',
+                       'default': None}
+                  ),
+                  'returns': {'type': ('null', 'dict'), 'desc': 'The AHA service information dictionary, or $lib.null.', }}},
+        {'name': 'list', 'desc': 'List the services registered with AHA.',
+         'type': {'type': 'function', '_funcname': '_methSvcList', 'args': (),
+                  'returns': {'type': 'list', 'desc': 'A list of the AHA service dictionaries for all registered services.', }}},
+    )
+    _storm_lib_path = ('aha',)
+    def getObjLocals(self):
+        return {
+            'del': self._methSvcDel,
+            'get': self._methSvcGet,
+            'list': self._methSvcList,
+        }
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _methSvcList(self):
+        self.runt.reqAdmin()
+        proxy = await self.runt.snap.core.reqAhaProxy()
+        # DISCUSS: Do we want this to be a generator or list?
+        svcs = []
+        async for info in proxy.getAhaSvcs():
+            svcs.append(info)
+        return svcs
+
+    async def _methSvcDel(self, name, network=None):
+        self.runt.reqAdmin()
+        name = await s_stormtypes.tostr(name)
+        network = await s_stormtypes.tostr(network, noneok=True)
+        proxy = await self.runt.snap.core.reqAhaProxy()
+        return await proxy.delAhaSvc(name, network=network)
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _methSvcGet(self, name, filters=None):
+        self.runt.reqAdmin()
+        name = await s_stormtypes.tostr(name)
+        filters = await s_stormtypes.toprim(filters)
+        proxy = await self.runt.snap.core.reqAhaProxy()
+        return await proxy.getAhaSvc(name, filters=filters)
+
 @s_stormtypes.registry.registerLib
 class AhaPoolLib(s_stormtypes.Lib):
     '''
     A Storm Library for interacting with AHA service pools.
     '''
+
+    _storm_locals = (
+        {'name': 'add', 'desc': 'Add a new AHA service pool.',
+         'type': {'type': 'function', '_funcname': '_add',
+                  'args': (
+                      {'name': 'name', 'type': 'str', 'desc': 'The name of the pool to add.', },
```

Comment:
Do we want to make note of these APIs being most easily used with relative AHA names?